lib/instrument/headers.js

```diff
@@ -0,0 +1,339 @@
+    return messageHeader;
+  }
+
+  static parse (headerBuffer) {
+    return {
+      flags: headerBuffer.readUInt32LE(0),
+      auxLength: headerBuffer.readUInt32LE(4),
+      totalLength: Number(headerBuffer.readBigInt64LE(8))
+    };
+  }
+}
+
+
+class DTXMessageAux {
+  constructor (data = []) {
+    this.data = data;
+  }
+
+  build () {
+    const messageAux = new DTXMessageAuxBuffer();
+    if (this.data.length === 0) {
+      return Buffer.alloc(0);
+    }
+    for (const arg of this.data) {
+      if (typeof arg === 'number') {
+        if (Math.abs(arg) < INT32) {
+          messageAux.appendInt(arg);
+        } else if (Math.abs(arg) < INT64) {
+          messageAux.appendLong(arg);
+        } else {
+          throw new Error('num too large');
+        }
+      } else {
+        messageAux.appendObject(arg);
+      }
+    }
+    return messageAux.getBytes();
+  }
+
+  static parse (headerBuffer) {
+    let cursor = 0;
+    // // magic not find the rule
+    // const magic = Number(headerBuffer.readBigInt64LE(0));
+    // cursor += 8;
+    // if (magic !== DTX_MESSAGE_AUX_MAGIC) {
+    //   throw new Error('incorrect aux header magic: ' + magic);
+    // }
+    const data = [];
+    const length = headerBuffer.readBigInt64LE(8);
+    cursor += 16;
+    while (cursor <= length) {
+      const m = headerBuffer.readUInt32LE(cursor);
+      if (m !== DTX_AUXILIARY_MAGIC) {
+        throw new Error('incorrect auxiliary magic: ' + m);
+      }
+      const type = headerBuffer.readUInt32LE(cursor + 4);
+      cursor += 8;
+      switch (type) {
+        case AUX_TYPES.NSKeyed: {
+          const strLen = headerBuffer.readUInt32LE(cursor);
+          cursor += 4;
+          const archived_data = headerBuffer.slice(cursor, cursor + strLen);
+          data.push(unarchive(archived_data));
+          cursor += strLen;
+          break;
+        }
+        case AUX_TYPES.UInt32LE:
+          data.push(headerBuffer.readUInt32LE(cursor));
+          cursor += 4;
+          break;
+        case AUX_TYPES.BigUInt64LE:
+          data.push(headerBuffer.readBigUInt64LE(cursor));
+          cursor += 8;
+          break;
+        case AUX_TYPES.BigInt64LE:
+          data.push(headerBuffer.readBigInt64LE(cursor));
+          cursor += 8;
+          break;
+      }
+    }
+    return data;
+  }
+}
+
+class InstrumentRPCParseError {
+  constructor (data) {
+    this.data = data;
+  }
+
+}
+
+class DTXMessageAuxBuffer {
+  constructor () {
+    this._buf = Buffer.alloc(0);
+  }
+
+  appendInt (value) {
+    const buf = Buffer.alloc(12);
+    buf.writeUInt32LE(10, 0);
+    buf.writeUInt32LE(3, 4);
+    buf.writeUInt32LE(value, 8);
+    this._buf = Buffer.concat([this._buf, buf]);
+
+  }
+
+  appendLong (value) {
+    const buf = Buffer.alloc(16);
+    buf.writeUInt32LE(10, 0);
+    buf.writeUInt32LE(4, 4);
+    // eslint-disable-next-line no-undef
+    buf.writeBigUInt64LE(BigInt(value), 8);
```

Comment:
this won't really work if the incoming value is not of type BigInt already

Reply:
I don't know how to write uInt64 in nodejs Buffer. This approach seems to work

Reply:
Using this method requires converting int to bigint , or what should i do？

Reply:
like I said above this works, but actually does not do anything. we should make sure that the incoming value has the correct type already (e.g. make a BigInt when it is read from Buffer). Check https://nodejs.org/api/buffer.html#bufreadbigint64leoffset for more details

Reply:
I know what you mean, but the totalLength is generated by the DTXMessageAuxBuffer and selectorBuffer lengths. convert bigint should be required.
```
totalLength: aux.length + sel.length,
```
I will modify other code about bigint

Reply:
Then you probably need to do some more magic based on the actual variable length (e.g. if it is more than 4 bytes that use bigint otherwise a vanilla number). Check https://stackoverflow.com/questions/52595180/how-to-read-write-a-bigint-from-buffer-in-node-js-10 for more details on how buffer values could be properly packed/unpacked into bigint